src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/handlers/presubmit/AsylumCaseNotificationHandler.java

```diff
@@ -0,0 +1,67 @@
+import uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.entities.MessageHeader;
+import uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.entities.ccd.Event;
+import uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.entities.ccd.callback.Callback;
+import uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.entities.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.entities.ccd.callback.PreSubmitCallbackStage;
+import uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.handlers.PreSubmitCallbackHandler;
+import uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.service.HomeOfficeInstructService;
+
+@Component
+public class AsylumCaseNotificationHandler implements PreSubmitCallbackHandler<AsylumCase> {
+
+    private HomeOfficeInstructService homeOfficeInstructService;
+
+    public AsylumCaseNotificationHandler(HomeOfficeInstructService homeOfficeInstructService) {
+        this.homeOfficeInstructService = homeOfficeInstructService;
+    }
+
+    @Override
+    public boolean canHandle(PreSubmitCallbackStage callbackStage, Callback<AsylumCase> callback) {
+        requireNonNull(callbackStage, "callbackStage must not be null");
+        requireNonNull(callback, "callback must not be null");
+
+        return callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT
+            && callback.getEvent() == Event.SUBMIT_APPEAL
+            && callback.getCaseDetails().getState() == APPEAL_SUBMITTED;
+    }
+
+    @Override
+    public PreSubmitCallbackResponse<AsylumCase> handle(
+        PreSubmitCallbackStage callbackStage, Callback<AsylumCase> callback) {
+
+        if (!canHandle(callbackStage, callback)) {
+            throw new IllegalStateException("Cannot handle callback");
+        }
+
+        AsylumCase asylumCase = callback.getCaseDetails().getCaseData();
+
+        final String homeOfficeReferenceNumber = asylumCase.read(HOME_OFFICE_REFERENCE_NUMBER, String.class)
+            .orElseThrow(() -> new IllegalStateException("Home office reference for the appeal is not present"));
+
+        final String caseId = asylumCase.read(APPEAL_REFERENCE_NUMBER, String.class)
+            .orElseThrow(() -> new IllegalStateException("Case ID for the appeal is not present"));
+
+        MessageHeader messageHeader = homeOfficeInstructService.sendNotification(homeOfficeReferenceNumber,caseId);
+        PreSubmitCallbackResponse<AsylumCase> response = new PreSubmitCallbackResponse<>(asylumCase);
+        if (messageHeader == null) {
+            asylumCase.write(HOME_OFFICE_INSTRUCT_STATUS,"Error sending notification to Home Office");
+            response.addErrors(Collections.singleton("Error sending notification to Home Office"));
+        } else {
+            asylumCase.write(HOME_OFFICE_INSTRUCT_STATUS,"Notification sent to Home Office");
+        }
+
+        return new PreSubmitCallbackResponse<>(asylumCase);
```

Comment:
So it looks like you've created a `new PreSubmitCallbackResponse` on line 56 and you are creating another one here?  Surely this is only required once?

Reply:
It is removed here and already fixed in #77

Reply:
please make sure you fix this here.  You are trying to merge this PR so should be fixed here.

Reply:
It is fixed in this PR